src/MassTransit/Configuration/Registration/RegistrationConfigurator.cs

```diff
@@ -311,14 +311,16 @@ protected static void ConfigureLogContext(IConfigurationServiceProvider provider
                 LogContext.ConfigureCurrentLogContext(loggerFactory);
         }
 
-        protected void ConfigureMediator(IReceiveEndpointConfigurator configurator, IConfigurationServiceProvider provider)
+        protected void ConfigureMediator(IReceiveEndpointConfigurator configurator, IConfigurationServiceProvider provider,
+            Action<IRegistration, IReceiveEndpointConfigurator> configure)
         {
             var registration = new Registration(provider, _consumerRegistrations.ToDictionary(x => x.Key, x => x.Value),
                 _sagaRegistrations.ToDictionary(x => x.Key, x => x.Value), _executeActivityRegistrations.ToDictionary(x => x.Key, x => x.Value),
                 _activityRegistrations.ToDictionary(x => x.Key, x => x.Value), _endpointRegistrations.ToDictionary(x => x.Key, x => x.Value));
 
             registration.ConfigureConsumers(configurator);
             registration.ConfigureSagas(configurator);
+            configure?.Invoke(registration, configurator);
```

Comment:
This needs to be before the ConfigureConsumers/Sagas method, since it may be configuring middleware.